LabBuilder/LabBuilder.psm1

```diff
@@ -1438,6 +1438,120 @@ function Get-LabVMTemplates {
+.DESCRIPTION
+   Takes a provided Lab Configuration file and returns the list of Template Disks
+   that will be used to create the Virtual Machines in this lab. This list is usually passed to
+   Initialize-LabVMTemplateDisks.
+.PARAMETER Configuration
+   Contains the Lab Builder configuration object that was loaded by the Get-LabConfiguration object.
+.EXAMPLE
+   $VMTemplateDisks = Get-LabVMTemplateDisks -Config c:\mylab\config.xml 
+   Loads a Lab Builder configuration and pulls the array of TemplateDisks from it.
+.OUTPUTS
+   Returns an array of TemplateDisks.
+#>
+function Get-LabVMTemplateDisks {
+    [OutputType([System.Collections.Hashtable[]])]
+    [CmdLetBinding()]
+    param
+    (
+        [Parameter(
+            Mandatory,
+            Position=0)]
+        [ValidateNotNullOrEmpty()]
+        [XML] $Config
+    )
+
+    [System.Collections.Hashtable[]] $VMTemplateDisks = @()
+    [String] $VHDParentPath = $Config.labbuilderconfig.SelectNodes('settings').vhdparentpath
+   
+	Write-Verbose "VHD parent path = $VHDParentPath" 
+   
+    # Read the list of templateDisks from the configuration file
+    $TemplateDisks = $Config.labbuilderconfig.SelectNodes('TemplateDisks').TemplateDisk
+    foreach ($TemplateDisk in $TemplateDisks)
+    {
+        # It can't be template because if the name attrib/node is missing the name property on
+        # the XML object defaults to the name of the parent. So we can't easily tell if no name
+        # was specified or if they actually specified 'template' as the name.
+        if ($TemplateDisk.Name -eq 'TemplateDisk')
+        {
+            $ExceptionParameters = @{
+                errorId = 'EmptyTemplateNameError'
+                errorCategory = 'InvalidArgument'
+                errorMessage = $($LocalizedData.EmptyTemplateNameError)
+            }
+            New-LabException @ExceptionParameters
+        } # If
+        
+        # Get the Template Disk Name
+        [String] $DiskName = ""
+        if ($TemplateDisk.name)
+        {
+            $DiskName = $TemplateDisk.Name
+        } # if
+        
+		Write-Verbose "Template disk name = $DiskName"
+
+        # Get the Template OS Type 
+        [String] $isNano = 'False'
+        if ($TemplateDisk.isNano)
+        {
+            $isNano = $TemplateDisk.isNano
+        } # If
+        
+		# Get the Template Wim Image to use 
+        [String] $WimImage = ""
+        if ($TemplateDisk.WimImage)
+        {
+            $WimImage = $TemplateDisk.WimImage
+        } # If
+
+        # Get the Template VHD Type 
+        [String] $VHDFormat = "VHDX"
+        if ($TemplateDisk.VHDFormat)
+        {
+            $VHDFormat = $TemplateDisk.VHDFormat
+        } # If
+
+        # Get the Template VM Generation 
+        [int] $Generation = "2"
+        if ($TemplateDisk.Generation)
+        {
+            $Generation = $TemplateDisk.Generation
+        } # If
+
+        # Get the Template Packages for Nano 
+        [String] $Packages = ""
```

Comment:
Nitpick: It is faster to use single quotes if you don't need advanced string merging/handling functions. Not a bit deal really, but code editors such as ISE Steroids while flag this as an issue.
e.g.
```powershell
[String] $Packages = ''
```